src/installer/managed/Microsoft.NET.HostModel/Bundle/Bundler.cs

```diff
@@ -44,22 +48,72 @@ public class Bundler
             OutputDir = Path.GetFullPath(string.IsNullOrEmpty(outputDir) ? Environment.CurrentDirectory : outputDir);
             Target = new TargetInfo(targetOS, targetArch, targetFrameworkVersion);
 
+            if (Target.BundleMajorVersion < 6 &&
+                (options & BundleOptions.EnableCompression) != 0)
+            {
+                throw new ArgumentException("Compression requires framework version 6.0 or above", nameof(options));
+            }
```

Comment:
Will we block this on SDK side as well? If not we need a better error message which uses SDK terminology.
But I would prefer to have a block in SDK itself which can provide better error message and context.

Reply:
Yes, we will have a proper block in SDK. 
Here we need to have something - assert or an argument check. It seems an argument check is more appropriate since it is at an entry point.

Reply:
Basically if we hit this it would be a bug in SDK or in a test.

Reply:
Got it - thanks!